golem-worker-executor-base/src/storage/keyvalue/sqlite.rs

```diff
@@ -0,0 +1,751 @@
+    }
+
+    async fn members_of_set(
+        &self,
+        svc_name: &'static str,
+        api_name: &'static str,
+        _entity_name: &'static str,
+        namespace: KeyValueStorageNamespace,
+        key: &str,
+    ) -> Result<Vec<Bytes>, String> {
+        self.pool
+            .with(svc_name, api_name)
+            .members_of_set(&Self::to_string(&namespace), key)
+            .await
+            .map_err(|e| e.to_string())
+    }
+
+    async fn add_to_sorted_set(
+        &self,
+        svc_name: &'static str,
+        api_name: &'static str,
+        _entity_name: &'static str,
+        namespace: KeyValueStorageNamespace,
+        key: &str,
+        score: f64,
+        value: &[u8],
+    ) -> Result<(), String> {
+        self.pool
+            .with(svc_name, api_name)
+            .add_to_sorted_set(&Self::to_string(&namespace), key, score, value)
+            .await
+            .map_err(|e| e.to_string())
+    }
+
+    async fn remove_from_sorted_set(
+        &self,
+        svc_name: &'static str,
+        api_name: &'static str,
+        _entity_name: &'static str,
+        namespace: KeyValueStorageNamespace,
+        key: &str,
+        value: &[u8],
+    ) -> Result<(), String> {
+        self.pool
+            .with(svc_name, api_name)
+            .remove_from_sorted_set(&Self::to_string(&namespace), key, value)
+            .await
+            .map_err(|e| e.to_string())
+    }
+
+    async fn get_sorted_set(
+        &self,
+        svc_name: &'static str,
+        api_name: &'static str,
+        _entity_name: &'static str,
+        namespace: KeyValueStorageNamespace,
+        key: &str,
+    ) -> Result<Vec<(f64, Bytes)>, String> {
+        self.pool
+            .with(svc_name, api_name)
+            .get_sorted_set(&Self::to_string(&namespace), key)
+            .await
+            .map_err(|e| e.to_string())
+    }
+
+    async fn query_sorted_set(
+        &self,
+        svc_name: &'static str,
+        api_name: &'static str,
+        _entity_name: &'static str,
+        namespace: KeyValueStorageNamespace,
+        key: &str,
+        min: f64,
+        max: f64,
+    ) -> Result<Vec<(f64, Bytes)>, String> {
+        self.pool
+            .with(svc_name, api_name)
+            .query_sorted_set(&Self::to_string(&namespace), key, min, max)
+            .await
+            .map_err(|e| e.to_string())
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct SqlitePool {
+    pool: SqlitePoolx,
+}
+
+impl SqlitePool {
+    pub async fn configured(config: &DbSqliteConfig) -> Result<Self, anyhow::Error> {
+        let pool = SqlitePoolOptions::new()
+            .max_connections(config.max_connections)
+            .connect(&config.database)
+            .await?;
+
+        SqlitePool::init(&pool).await?;
+
+        Ok(SqlitePool { pool })
+    }
+
+    async fn init(pool: &SqlitePoolx) -> Result<(), Error> {
+        sqlx::query(
+            r#"
+            CREATE TABLE IF NOT EXISTS kv_storage (
```

Comment:
is there reason why this is not in db migrations script but in code ?

Reply:
Not sure how it would look like with db migration scripts, but I would like to have these storage trait implementations self contained - so you can drop in different implementations without any external requirements.

Reply:
I think the idea is to allows the database to be automatically initialise when the application starts, without needing a separate step to run migration scripts, which is faster and without external dependencies.
For this small-scale sqlite usage (for POCs or local development), and  without having complex schema definitions, I think migrations aren’t needed.
Probably we can go with this approach and I think we can later on add the migration part if needed.